kolibri/plugins/facility/assets/src/views/ManageClassPage/ClassCopyModal.vue

```diff
@@ -0,0 +1,238 @@
+        />
+        <KCheckbox
+          v-if="classCoachesIds.length"
+          :checked="copyAllCoaches"
+          :label="copyAllCoaches$({ n: classCoachesIds.length })"
+          @change="copyAllCoaches = $event"
+        />
+      </div>
+    </div>
+    <template #actions>
+      <KButtonGroup>
+        <KButton
+          :text="coreString('cancelAction')"
+          :disabled="loading"
+          @click="$emit('close')"
+        />
+        <KButton
+          primary
+          :text="makeACopy$()"
+          :disabled="Boolean(classNameInvalidText) || loading"
+          @click="copyClass"
+        />
+      </KButtonGroup>
+    </template>
+  </KModal>
+
+</template>
+
+
+<script>
+
+  import { computed, getCurrentInstance, onMounted, ref } from 'vue';
+  import useSnackbar from 'kolibri/composables/useSnackbar';
+  import commonCoreStrings from 'kolibri/uiText/commonCoreStrings';
+  import { UserKinds } from 'kolibri/constants';
+  import ClassroomResource from 'kolibri-common/apiResources/ClassroomResource';
+  import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResource';
+  import MembershipResource from 'kolibri-common/apiResources/MembershipResource';
+  import RoleResource from 'kolibri-common/apiResources/RoleResource';
+  import { bulkUserManagementStrings } from 'kolibri-common/strings/bulkUserManagementStrings';
+
+  export default {
+    name: 'ClassCopyModal',
+    mixins: [commonCoreStrings],
+    setup(props, context) {
+      const classCoachesIds = ref([]);
+      const classLearnerIds = ref([]);
+      const loading = ref(false);
+      const copyAllLearners = ref(false);
+      const copyAllCoaches = ref(false);
+      const createdClass = ref(null);
+      const copiedClassName = ref(null);
+      const store = getCurrentInstance().proxy.$store;
+      const {
+        copyClass$,
+        makeACopy$,
+        copyOfClass$,
+        copyAllCoaches$,
+        copyAllLearners$,
+        defaultErrorMessage$,
+        classNameAlreadyExists$,
+        classCopiedSuccessfully$,
+      } = bulkUserManagementStrings;
+      const { coreString } = commonCoreStrings.methods;
+      const { createSnackbar } = useSnackbar();
+      const { classToCopy, classes } = props;
+
+      const classNameInvalidText = computed(() => {
+        if (!loading.value) {
+          const name = (copiedClassName.value || '').trim();
+          if (!name) {
+            return coreString('requiredFieldError');
+          } else if (classes.some(row => row.name === name)) {
+            return classNameAlreadyExists$({ class: name });
+          }
+        }
+        return '';
+      });
+
+      // methods
+      function closeWithSnackbar(message) {
+        context.emit('close');
+        createSnackbar(message);
+      }
+
+      function handleApiFailure(error) {
+        closeWithSnackbar(defaultErrorMessage$());
+        store.dispatch('handleApiError', { error }, { root: true });
+      }
```

Comment:
We don't need to tell the user there was an error in two different ways here. It probably be a good idea if we were giving a non-default and more actionable error message in the snackbar but in this case, being taken to the handleApiError screen is probably sufficient.